keras/applications/imagenet_utils.py

```diff
@@ -18,26 +18,32 @@ def _preprocess_numpy_input(x, data_format, mode):
         x -= 1.
         return x
 
-    if data_format == 'channels_first':
-        if x.ndim == 3:
+    if mode == 'torch':
+        x /= 255.
+        mean = [0.485, 0.456, 0.406]
+        std = [0.229, 0.224, 0.225]
+    else:
+        if data_format == 'channels_first':
             # 'RGB'->'BGR'
-            x = x[::-1, ...]
-            # Zero-center by mean pixel
-            x[0, :, :] -= 103.939
-            x[1, :, :] -= 116.779
-            x[2, :, :] -= 123.68
+            if x.ndim == 3:
+                x = x[::-1, ...]
+            else:
+                x = x[:, ::-1, ...]
         else:
-            x = x[:, ::-1, ...]
-            x[:, 0, :, :] -= 103.939
-            x[:, 1, :, :] -= 116.779
-            x[:, 2, :, :] -= 123.68
-    else:
-        # 'RGB'->'BGR'
-        x = x[..., ::-1]
-        # Zero-center by mean pixel
-        x[..., 0] -= 103.939
-        x[..., 1] -= 116.779
-        x[..., 2] -= 123.68
+            # 'RGB'->'BGR'
+            x = x[..., ::-1]
+        mean = [103.939, 116.779, 123.68]
+        std = None
+
+    if data_format == 'channels_first':
+        mean = np.expand_dims(np.expand_dims(mean, axis=1), axis=2)
+        if std is not None:
+            std = np.expand_dims(np.expand_dims(std, axis=1), axis=2)
```

Comment:
The problem of this approach is that it instantiates (and reshapes) a Numpy array at every call. Since this method is meant to be called in a loop, it will end up being expensive. it would be preferable to follow the previous pattern to avoid this.

Reply:
I agree. Reverted.